packages/core/src/services/evaluationResults/fetchEvaluationResultsByDocumentLogs.ts

```diff
@@ -0,0 +1,45 @@
+import { database } from '../../client'
+import { Result } from '../../lib'
+import {
+  EvaluationResultsRepository,
+  EvaluationsRepository,
+  ResultWithEvaluation,
+} from '../../repositories'
+
+export async function fetchEvaluationResultsByDocumentLogs(
+  {
+    workspaceId,
+    documentLogIds,
+  }: {
+    workspaceId: number
+    documentLogIds: number[]
+  },
+  db = database,
+) {
+  documentLogIds = [...new Set(documentLogIds)]
+
+  const evaluationsRepository = new EvaluationsRepository(workspaceId, db)
+  const resultsRepository = new EvaluationResultsRepository(workspaceId, db)
+
+  const results = await resultsRepository
+    .findByDocumentLogIds(documentLogIds)
+    .then((r) => r.unwrap())
+
+  const evaluations = await evaluationsRepository
+    .filterById([...new Set(results.map((r) => r.evaluationId))])
```

Comment:
Why a Set here?

Reply:
To avoid getting unnecessary duplicated evaluations if there are various results of the same evaluation.